clients/src/main/java/org/apache/kafka/common/network/SaslChannelBuilder.java

```diff
@@ -52,7 +55,15 @@ public SaslChannelBuilder(Mode mode, LoginType loginType, SecurityProtocol secur
     public void configure(Map<String, ?> configs) throws KafkaException {
         try {
             this.configs = configs;
-            this.loginManager = LoginManager.acquireLoginManager(loginType, configs);
+            String mechanism = (String) this.configs.get(SaslConfigs.SASL_MECHANISM);
+            if (mechanism == null)
+                throw new IllegalArgumentException("SASL mechanism not specified.");
+            boolean hasKerberos = mechanism.equals(SaslConfigs.GSSAPI_MECHANISM);
+            if (mode == Mode.SERVER) {
+                List<String> enabledMechanisms = (List<String>) this.configs.get(SaslConfigs.SASL_ENABLED_MECHANISMS);
+                if (enabledMechanisms != null && !hasKerberos)
```

Comment:
@junrao You are right, server doesn't need to check the SASL mechanism. But it can disable Kerberos if GSSAPI is not included in `enabledMechanisms`. Have updated the check.